src/polymer-language-server.ts

```diff
@@ -174,75 +202,58 @@ async function autoComplete(textPosition: TextDocumentPositionParams):
-      // that
-      // starts with e.g. `C:\`. We can normalize these two by forcing both into
-      // an absolute path, like so:
-      const absolutePath = path.resolve('/', decodedPath);
-      // But what the editor service really wants is a path relative to the
-      // workspace root. So do that.
-      return path.relative(workspaceRoot, absolutePath);
-    }
+function getWorkspacePathToFile(doc: {uri: string}): string|undefined {
+  const docUrl = url.parse(doc.uri);
+  if (!workspaceUri || !workspaceUri.pathname || !docUrl.pathname) {
+    return undefined;
+  }
+  const workspacePath = fileUrlToAbsolutePath(workspaceUri);
+  const docPath = fileUrlToAbsolutePath(docUrl);
+  return path.relative(workspacePath, docPath);
+}
 
-function
-getUriForLocalPath(localPath: string):
-    string {
-      return `file://${workspaceRoot}/${localPath}`;
-    }
+function getUriForLocalPath(localPath: string): string {
+  if (!workspaceUri) {
+    throw new Error(`Tried to get the URI of ${localPath
+                    } without knowing the workspaceUri!?`);
+  }
+  const workspacePath = fileUrlToAbsolutePath(workspaceUri);
+  const absolutePath = path.join(workspacePath, localPath);
+  return fileUrl(absolutePath);
+}
 
-function
-convertPosition(position: LSPosition):
-    SourcePosition {
-      return {line: position.line, column: position.character};
-    }
+function convertPosition(position: LSPosition): SourcePosition {
+  return {line: position.line, column: position.character};
+}
 
-function
-convertRange(range: SourceRange):
-    Range {
-      return {
-        start: {line: range.start.line, character: range.start.column},
-        end: {line: range.end.line, character: range.end.column}
-      };
-    }
+function convertRange(range: SourceRange): Range {
+  return {
+    start: {line: range.start.line, character: range.start.column},
+    end: {line: range.end.line, character: range.end.column}
+  };
+}
 
-function
-convertSeverity(severity: Severity):
-    DiagnosticSeverity {
-      switch (severity) {
-        case Severity.ERROR:
-          return DiagnosticSeverity.Error;
-        case Severity.WARNING:
-          return DiagnosticSeverity.Warning;
-        case Severity.INFO:
-          return DiagnosticSeverity.Information;
-        default:
-          throw new Error(
-              `This should never happen. Got a severity of ${severity}`);
-      }
-    }
+function convertSeverity(severity: Severity): DiagnosticSeverity {
+  switch (severity) {
+    case Severity.ERROR:
+      return DiagnosticSeverity.Error;
+    case Severity.WARNING:
+      return DiagnosticSeverity.Warning;
+    case Severity.INFO:
+      return DiagnosticSeverity.Information;
+    default:
+      throw new Error(
+          `This should never happen. Got a severity of ${severity}`);
+  }
+}
 
-function
-scanDocument(document: TextDocument, connection?: IConnection) {
+function scanDocument(document: TextDocument, connection?: IConnection) {

```

Comment:
can you document that this is async? Showing that it returns a promise or making async would probably be the easiest way to do this